src/plugins/WebOfTrust/util/TickerDelayedBackgroundJob.java

```diff
@@ -0,0 +1,363 @@
+package plugins.WebOfTrust.util;
+
+import freenet.node.FastRunnable;
+import freenet.node.PrioRunnable;
+import freenet.support.Executor;
+import freenet.support.Ticker;
+import freenet.support.io.NativeThread;
+
+/**
+ * A {@link DelayedBackgroundJob} implementation that uses a {@link Ticker} for scheduling.
+ *
+ * @author bertm
+ * @see TickerDelayedBackgroundJobFactory
+ */
+public class TickerDelayedBackgroundJob implements DelayedBackgroundJob {
+    static enum JobState {
+        /** Waiting for a trigger, no running job thread or scheduled job. */
+        IDLE,
+        /** Waiting for the scheduled job to be executed, no running job thread. */
+        WAITING,
+        /** Running the job in the job thread. */
+        RUNNING,
+        /** Waiting for the running job thread to finish before we terminate. */
+        TERMINATING,
+        /** Terminated, no running job thread. */
+        TERMINATED
+    }
+
+    /** Constant for {@link #nextExecutionTime} meaning there is no next execution requested. */
+    private final static long NO_EXECUTION = Long.MAX_VALUE;
+
+    /** Job wrapper for status tracking. */
+    private final DelayedBackgroundRunnable realJob;
+    /** Human-readable name of this job. */
+    private final String name;
+    /** Aggregation delay in milliseconds. */
+    private final long defaultDelay;
+    /** The ticker used to schedule this job. */
+    private final Ticker ticker;
+    /** The executor of this background job. */
+    private final Executor executor;
+
+    /** Tiny job to run on the ticker, invoking the execution of the {@link #realJob} on the
+     * executor (only set in {@code WAITING} state). */
+    private Runnable waitingTickerJob = null;
+    /** Running state of this job. */
+    private JobState state = JobState.IDLE;
+    /** Job execution thread, only if the job is running (only set in {@code RUNNING} state). */
+    private Thread thread = null;
+    /** Next absolute time we have a job execution scheduled, or {@link #NO_EXECUTION} if none. */
+    private long nextExecutionTime = NO_EXECUTION;
+
+    /**
+     * Constructs a delayed background job with the given default delay. Negative delays
+     * are treated as zero delay.
+     * The {@link Ticker} given and its {@link Executor} <b>must</b> have an asynchronous
+     * implementation of respectively
+     * {@link Ticker#queueTimedJob(Runnable, String, long, boolean, boolean) queueTimedJob} and
+     * {@link Executor#execute(Runnable, String) execute}. When this background job is
+     * {@link BackgroundJob#terminate() terminated}, the running job will be notified by means of
+     * interruption of its thread. Hence, the job implementer must take care not to swallow
+     * {@link InterruptedException}.
```

Comment:
This is also subject to the JavaDoc improment suggested here:
https://github.com/freenet/plugin-WebOfTrust/pull/26/files#r22342146

There's a lot of constructor JavaDoc copypaste which needs this already. I've done a short Google search, and unfortunately it seems like there is no  {@inheritDoc} for constructors. @see or @link could do the job, maybe something like: "You <b>must</b> also follow the requirements of the parent constructor {@link ...}."